glass/src/lib/native/cpp/Window.cpp

```diff
@@ -65,10 +65,27 @@ void Window::Display() {
   }
 
   if (Begin(label.c_str(), &m_visible, m_flags)) {
+    ImGuiWindow* window = ImGui::GetCurrentWindow();
+    // Constrains the window within the bounds of the viewport, resizing as
+    // necessary
+    ImVec2 min = {};
+    ImVec2 max = ImGui::GetMainViewport()->Size;
+    ImVec2 newSize = window->Size;
+    ImVec2 newPos = window->Pos;
+    // Might need to shrink window to fit in viewport
+    if (max.x - window->Size.x < min.x) {
+      newSize.x = max.x;
+    }
+    if (max.y - window->Size.y < min.y) {
+      newSize.y = max.y;
+    }
+    ImGui::SetWindowSize(newSize);
+    newPos.x = std::clamp(window->Pos.x, min.x, max.x - newSize.x);
+    newPos.y = std::clamp(window->Pos.y, min.y, max.y - newSize.y);
+    ImGui::SetWindowPos(newPos);
```

Comment:
```suggestion
    ImVec2 newSize = {std::clamp(window->Size.x, 0, max.x - min.x), std::clamp(window->Size.y, 0, max.y - min.y)};
    ImGui::SetWindowSize(newSize);
    ImVec2 newPos = {std::clamp(window->Pos.x, min.x, max.x - newSize.x), std::clamp(window->Pos.y, min.y, max.y - newSize.y)};
    ImGui::SetWindowPos(newPos);
```
Note that the maximum value we want for the size is max - min. (A nice gut check is that you can rearrange `max - size < min` into `max - min < size` or `size > max - min`)